x/stake/keeper/delegation.go

```diff
@@ -81,7 +81,7 @@ func (k Keeper) SetDelegation(ctx sdk.Context, delegation types.Delegation) {
 
 // remove a delegation from store
 func (k Keeper) RemoveDelegation(ctx sdk.Context, delegation types.Delegation) {
-	k.OnDelegationRemoved(ctx, delegation.DelegatorAddr, delegation.ValidatorAddr)
+	k.PostDelegationRemoved(ctx, delegation.DelegatorAddr, delegation.ValidatorAddr)
```

Comment:
This isn't post-modification (as you can see by the `store.Delete` call two lines below)